src/js/common/schemaform/FormApp.jsx

```diff
@@ -43,14 +56,40 @@ export default class FormApp extends React.Component {
     window.removeEventListener('beforeunload', this.onbeforeunload);
   }
 
+  resetLoading = () => {
+    this.props.setFetchFormStatus(LOAD_STATUSES.notAttempted);
+  }
+
+  resumeForm = () => {
+    this.props.fetchInProgressForm(this.props.formConfig.formId, this.props.formConfig.migrations);
+  }
+
+  goToBeginning = () => {
+    this.props.router.push(this.props.route.pageList[1].path);
+  }
+
   render() {
     const { currentLocation, formConfig, children } = this.props;
     const trimmedPathname = currentLocation.pathname.replace(/\/$/, '');
     const isIntroductionPage = trimmedPathname.endsWith('introduction');
     const isConfirmationPage = trimmedPathname.endsWith('confirmation');
     const GetFormHelp = formConfig.getHelp;
     let content;
-    if (!isInProgress(trimmedPathname)) {
+
+    // TODO: As soon as we've redirected, remember to set loadedStatus back to 'not-attempted'
+    if (!formConfig.disableSave && this.props.loadedStatus !== LOAD_STATUSES.notAttempted) {
+      // Show the loading screen instead of the children.
+      // Agh! The prop threading!
```

Comment:
I think part of the reason this is painful is because `LoadingPage` is trying to be two things: a separate page for the user to figure out what to do on an error, and a spinner indicating that there's a pending state. The way it is now is sort of confusing, because it feels like you're on a new page when there's an error, but the url hasn't changed.

I think you can split this in two. Make `LoadingPage` something like `ErrorPage` and remove the spinner. You can add it to the routes for the app. Then make a `InProgressFormLoading` component, that `FormApp` can put up whenever it sees the right loading status (or just use `LoadingIndicator`). `FormApp` would probably also handle routing to the `ErrorPage` or the return url, based on the load status, which I think is ok.

Prop threading would go away, because `ErrorPage` would be a connected component and `FormApp` would really just handling some global routing, which seems ok since it's the global form component.